uxarray/core/dataarray.py

```diff
@@ -97,3 +96,94 @@ def uxgrid(self):
+        representing the geometry of the unstructured grid, and a data column
+        representing a 1D slice of data mapped to each Polygon.
+
+        Parameters
+        Returns
+        -------
+        gdf : spatialpandas.GeoDataFrame
+            The output `GeoDataFrame` with a filled out "geometry" and 1D data column representing the geometry of the unstructured grid
+        """
+
+        # data is multidimensional, must be a 1D slice
+        if self.data.ndim > 1:
+            raise ValueError(
+                f"Data Variable must be 1-dimensional, with shape {self.uxgrid.nMesh2_face} "
+                f"for face-centered data.")
+
+        # face-centered data
+        if self.data.size == self.uxgrid.nMesh2_face:
+            gdf = self.uxgrid.to_geodataframe(override=override_grid,
+                                              cache=cache_grid)
+            gdf[self.name] = self.data
+            return gdf
+
+        # TODO: Mapping Node Data to Each Polygon
+        elif self.data.size == self.uxgrid.nMesh2_node:
+            raise ValueError(
+                f"Data Variable with size {self.data.size} mapped on the nodes of each polygon"
+                f"not supported yet.")
+
+        # data not mapped to faces or nodes
+        else:
+            raise ValueError(
+                f"Data Variable with size {self.data.size} does not match the number of faces "
+                f"({self.uxgrid.nMesh2_face}.")
+
+    def to_polycollection(self, override_grid=False, cache_grid=True):
+        """Constructs a ``matplotlib.collections.PolyCollection`` object with
+        polygons representing the geometry of the unstructured grid, with
+        polygons that cross the antimeridian split across the antimeridian.
+
+        Parameters
+        ----------
+        override_grid : bool
+            Flag to recompute the ``PolyCollection`` if one is already cached
+        cache_grid : bool
+            Flag to indicate if the computed ``PolyCollection`` should be cached
+
+        Returns
+        -------
+        poly_collection : matplotlib.collections.PolyCollection
+            The output `PolyCollection` of polygons representing the geometry of the unstructured grid paired with
+            a data variable.
+        """
+
+        # data is multidimensional, must be a 1D slice
+        if self.data.ndim > 1:
+            raise ValueError(
+                f"Data Variable must be 1-dimensional, with shape {self.uxgrid.nMesh2_face} "
+                f"for face-centered data.")
+
+        # face-centered data
+        if self.data.size == self.uxgrid.nMesh2_face:
+            poly_collection = self.uxgrid.to_polycollection(
+                override=override_grid, cache=cache_grid)
+
+            # map data with antimeridian polygons
+            if self.uxgrid.corrected_polygon_shells is not None:
+                data = self.data[self.uxgrid.original_to_corrected_indices]
```

Comment:
```suggestion
                data = self.data[self.uxgrid.corrected_shells_to_original_faces]
```
I think this just needs to be updated to reflect the name change